cfile/cfile.cpp

```diff
@@ -69,6 +79,112 @@ cfile_error cfe;
 // The message for unexpected end of file
 const char *eof_error = "Unexpected end of file";
 
+// This function must be called before you use anything else from this module.
+void cf_Init(std::vector<std::filesystem::path> initial_base_directories) {
+  Base_directories = initial_base_directories;
+}
```

Comment:
I made the requested change. I decided to call the function `cf_AddBaseDirectory()` because I really dislike the name `cf_AddSearchPath()`. The variable is called `Base_directories`, not `Search_paths`, so the name `cf_AddBaseDirectory()` is more clear. Additionally, there’s already two functions that are declared in the same file that are named `cf_SetSearchPath()` and `cf_ClearAllSearchPaths()`, and those two functions don’t change the `Base_directories` variable. It would be really confusing if `cf_AddSearchPath()` affected `Base_directories` but `cf_SetSearchPath()` did not.